clippy.toml

```diff
@@ -12,5 +12,6 @@ disallowed-methods = [
     { path = "crossbeam_channel::unbounded", reason = "Using an unbounded channel most likely will read to unbounded memory growth. Please use a bounded channel instead. If you are intentionally using unbounded channel, use `#[allow(clippy::disallowed_methods)]` to locally disable this check." },
 ]
 disallowed-types = [
-    { path = "tokio::sync::Mutex", reason = "You should only use an asynchronous lock if you need to .await something while the lock is locked. Usually, this is not necessary, and you should avoid using an asynchronous lock when you can. Asynchronous locks are a lot slower than blocking locks. Please read how to share state effectively across async tasks https://tokio.rs/tokio/tutorial/shared-state. If you are intentionally using an async mutex, use `#[allow(clippy::disallowed_types)]` to locally disable this check." },
+    { path = "tokio::sync::Mutex", reason = "You should only use an asynchronous lock if you need to .await something while the lock is locked. Usually, this is not necessary, and you should avoid using an asynchronous lock when you can. Asynchronous locks are a lot slower than blocking locks. Please read how to share state effectively across async tasks https://tokio.rs/tokio/tutorial/shared-state. If you are intentionally using an async Mutex, use `#[allow(clippy::disallowed_types)]` to locally disable this check." },
```

Comment:
You might want to make the reason more prominent. E.g.

> This is slow (compared to what? by how much?). For (better) alternatives, see https://tokio.rs/tokio/tutorial/shared-state. This ban can be overridden by adding `#![allow(clippy::disallowed_types)] // This is an appropriate use of Mutex, because...` to the top of the file where the banned type is used.

I think the rest is not very helpful. E.g. "This usually is not necessary" is not actionable. Also, "You should only use X if you need Y." probably does not help, because presumably, the reason X is being used is that the author believes that they need Y.

Reply:
"the reason X is being used is that the author believes that they need Y"

Usually the reason is because the author copied the code. People use the first thing they believe it works without thinking about use cases or alternatives. So it is better not to give much options.